packages/flutter/lib/src/material/autocomplete.dart

```diff
@@ -0,0 +1,256 @@
+
+  static Widget _defaultFieldViewBuilder(BuildContext context, TextEditingController textEditingController, FocusNode focusNode, VoidCallback onFieldSubmitted) {
+    return _AutocompleteField(
+      focusNode: focusNode,
+      textEditingController: textEditingController,
+      onFieldSubmitted: onFieldSubmitted,
+    );
+  }
+
+  @override
+  _AutocompleteState<T> createState() => _AutocompleteState<T>();
+}
+
+class _AutocompleteState<T extends Object> extends State<Autocomplete<T>> {
+  @override
+  Widget build(BuildContext context) {
+    return RawAutocomplete<T>(
+      displayStringForOption: widget.displayStringForOption,
+      fieldViewBuilder: widget.fieldViewBuilder,
+      optionsBuilder: widget.optionsBuilder,
+      optionsViewBuilder: widget.optionsViewBuilder ?? (BuildContext context, AutocompleteOnSelected<T> onSelected, Iterable<T> options) {
+        return _AutocompleteOptions<T>(
+          displayStringForOption: widget.displayStringForOption,
+          onSelected: onSelected,
+          options: options,
+        );
+      },
+      onSelected: widget.onSelected,
+    );
+  }
+}
+
+// The default Material-style Autocomplete text field.
+class _AutocompleteField extends StatelessWidget {
+  const _AutocompleteField({
+    Key? key,
+    required this.focusNode,
+    required this.textEditingController,
+    required this.onFieldSubmitted,
+  }) : super(key: key);
+
+  final FocusNode focusNode;
+
+  final VoidCallback onFieldSubmitted;
+
+  final TextEditingController textEditingController;
+
+  @override
+  Widget build(BuildContext context) {
+    return TextFormField(
+      controller: textEditingController,
+      focusNode: focusNode,
+      onFieldSubmitted: (String value) {
+        onFieldSubmitted();
+      },
+    );
+  }
+}
+
+// The default Material-style Autocomplete options.
+class _AutocompleteOptions<T extends Object> extends StatelessWidget {
+  const _AutocompleteOptions({
+    Key? key,
+    required this.displayStringForOption,
+    required this.onSelected,
+    required this.options,
+  }) : super(key: key);
+
+  final AutocompleteOptionToString<T> displayStringForOption;
+
+  final AutocompleteOnSelected<T> onSelected;
+
+  final Iterable<T> options;
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      alignment: Alignment.topLeft,
+      child: Material(
+        elevation: 4.0,
+        child: Container(
+          height: 200.0,
+          child: ListView.builder(
+            padding: const EdgeInsets.all(8.0),
+            itemCount: options.length,
+            itemBuilder: (BuildContext context, int index) {
+              final T option = options.elementAt(index);
+              return GestureDetector(
+                onTap: () {
+                  onSelected(option);
+                },
+                child: ListTile(
+                  title: Text(displayStringForOption(option)),
+                ),
+              );
```

Comment:
What about: 
```suggestion
              return ListTile(
                onTap: () {
                  onSelected(option);
                },              
                title: Text(displayStringForOption(option)),
              );
```

This gives it an ink splash so there is some responsiveness for users to confirm that their action has some effect. This would be similar to Angular's autocomplete: https://material.angular.io/components/autocomplete/overview. However, I don't know if this accounts for the highlight on hover effect on desktop/web, which would be pretty important.

Edit: I wonder if ListTile is the correct solution for this, since the dropdown selection font size is by default larger than the text field font size, which looks a bit weird. Could we just use an `InkWell` or `InkResponse` widget here, and have `onHover` defined. Then, we could use the menu theming for defining the padding for each menu item based on platform: https://material.io/components/menus#specs

Reply:
Thanks for the great ideas.  Here it is with InkWell on Mac and iOS.  I'm hovering "elephant" on Mac.

<img width="1277" alt="Screen Shot 2021-01-21 at 11 06 09 AM" src="https://user-images.githubusercontent.com/389558/105399755-e7813400-5bd8-11eb-80d6-2bfa45f50362.png">